src/lv_widgets/lv_chart.c

```diff
@@ -303,43 +303,44 @@ void lv_chart_set_point_count(lv_obj_t * chart, uint16_t point_cnt)
     if(point_cnt < 1) point_cnt = 1;
 
     _LV_LL_READ_BACK(ext->series_ll, ser) {
-        if(ser->start_point != 0) {
-            lv_coord_t * new_points = lv_mem_alloc(sizeof(lv_coord_t) * point_cnt);
-            LV_ASSERT_MEM(new_points);
-            if(new_points == NULL) return;
-
-            if(point_cnt >= point_cnt_old) {
-                for(i = 0; i < point_cnt_old; i++) {
-                    new_points[i] =
-                        ser->points[(i + ser->start_point) % point_cnt_old]; /*Copy old contents to new array*/
-                }
-                for(i = point_cnt_old; i < point_cnt; i++) {
-                    new_points[i] = def; /*Fill up the rest with default value*/
-                }
-            }
-            else {
-                for(i = 0; i < point_cnt; i++) {
-                    new_points[i] =
-                        ser->points[(i + ser->start_point) % point_cnt_old]; /*Copy old contents to new array*/
-                }
-            }
-
-            /*Switch over pointer from old to new*/
-            lv_mem_free(ser->points);
-            ser->points = new_points;
-        }
-        else {
-            ser->points = lv_mem_realloc(ser->points, sizeof(lv_coord_t) * point_cnt);
-            LV_ASSERT_MEM(ser->points);
-            if(ser->points == NULL) return;
-            /*Initialize the new points*/
-            if(point_cnt > point_cnt_old) {
-                for(i = point_cnt_old - 1; i < point_cnt; i++) {
-                    ser->points[i] = def;
-                }
-            }
-        }
-
+    	if( !ser->ext_buf_assigned ) {
+			if(ser->start_point != 0) {
+				lv_coord_t * new_points = lv_mem_alloc(sizeof(lv_coord_t) * point_cnt);
+				LV_ASSERT_MEM(new_points);
+				if(new_points == NULL) return;
+
+				if(point_cnt >= point_cnt_old) {
+					for(i = 0; i < point_cnt_old; i++) {
+						new_points[i] =
+							ser->points[(i + ser->start_point) % point_cnt_old]; /*Copy old contents to new array*/
+					}
+					for(i = point_cnt_old; i < point_cnt; i++) {
+						new_points[i] = def; /*Fill up the rest with default value*/
+					}
+				}
+				else {
+					for(i = 0; i < point_cnt; i++) {
+						new_points[i] =
+							ser->points[(i + ser->start_point) % point_cnt_old]; /*Copy old contents to new array*/
+					}
+				}
+
+				/*Switch over pointer from old to new*/
+				lv_mem_free(ser->points);
+				ser->points = new_points;
+			}
+			else {
+				ser->points = lv_mem_realloc(ser->points, sizeof(lv_coord_t) * point_cnt);
+				LV_ASSERT_MEM(ser->points);
+				if(ser->points == NULL) return;
+				/*Initialize the new points*/
+				if(point_cnt > point_cnt_old) {
+					for(i = point_cnt_old - 1; i < point_cnt; i++) {
+						ser->points[i] = def;
+					}
+				}
+			}
+    	}
```

Comment:
Can you add an `LV_LOG_WARN` in the else branch to indicate that nothing will happen?

Reply:
Hi @kisvegabor ,

Sorry about missing the signal handler, it's just my current unfamiliarity with the internals of LVGL.  I have made a change to that now.

Before I update the pull request can I ask where you want the LV_LOG_WARN adding please, it isn't clear to me?

Kind Regards,

Pete

Reply:
> Sorry about missing the signal handler, it's just my current unfamiliarity with the internals of LVGL. I have made a change to that now.

No problem, this part wasn't even there 2 hours ago :slightly_smiling_face: 
I mean something like this in `lv_chart_set_point_count`:
```c
if( !ser->ext_buf_assigned ) {
 ...
} else {
  LV_LOG_WARN("Can't set the point coint if external array is used");
}
```

Reply:
I see, but currently it is still setting the point count even with the external buffer.   I am likely to use this to chart various densities of points from the same buffer. Looking through the code it should not cause any problem I can see.  Is that okay?  If not I can just move the point count setting into the conditional and add the else with the log.  I could just do another call to assign buffer function for my purposes if you would rather do things that way....

 What do you think?

Reply:
Upps, sorry. The bottom of the function was hidden in GitHub's diff and didn't notice ` ext->point_cnt = point_cnt;` there.

All good from my side.